docs/02-local-development.md

```diff
@@ -0,0 +1,245 @@
+# Local Development
+
+This guide explains how to set up and develop this application locally using the monorepo structure with npm workspaces.
+
+## Monorepo Structure
+
+This repository is organized as a **monorepo** using [npm workspaces](https://docs.npmjs.com/cli/v7/using-npm/workspaces). A monorepo allows multiple related packages to be managed in a single repository, making it easier to share code and maintain consistency.
+
+### Package Structure
+
+The repository contains two main workspaces in the `packages-v1` directory:
+
+```
+azure-typescript-langchainjs/
+├── packages-v1/
+│   ├── langgraph-agent/     # Core agent implementation
+│   └── server-api/           # Fastify API server
+├── package.json              # Root package with workspace config
+└── azure.yaml                # Azure deployment configuration
+```
+
+#### langgraph-agent
+
+The core agent implementation package containing:
+- **LangGraph agent architecture** for orchestrating AI components
+- **Azure OpenAI integration** for embeddings and completions
+- **Azure AI Search vector store** integration
+- **PDF document processing** and loading scripts
+- **Reusable utilities** for Azure credential management
+
+#### server-api
+
+The API server package that:
+- Exposes the agent functionality via REST API
+- Uses [Fastify](https://www.fastify.io/) web framework
+- Depends on the `langgraph-agent` package
+- Handles HTTP requests to the `/answer` endpoint
+
+### Separation of Concerns
+
+Each workspace has a clear responsibility:
+- **langgraph-agent**: Business logic, AI orchestration, data processing
+- **server-api**: HTTP layer, request/response handling, API contracts
+
+This separation makes it easy to:
+- Test agent logic independently
+- Reuse the agent in different contexts (CLI, API, batch processing)
+- Scale different components independently
+
+## Setting Up Local Development
+
+### 1. Install Dependencies
+
+From the repository root:
+
+```bash
+# Install all dependencies for all workspaces
+npm install
+```
+
+This will install dependencies for both the root project and all workspace packages.
+
+### 2. Configure Environment Variables
+
+Create a `.env` file in the repository root:
```

Comment:
The `az up` deployment creates the `.env` - the reader should complete that step instead of creating their own .env.